samples/test/routines.test.js

```diff
@@ -0,0 +1,132 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+const {assert} = require('chai');
+const {describe, it, before, after} = require('mocha');
+const cp = require('child_process');
+const uuid = require('uuid');
+
+const {BigQuery} = require('@google-cloud/bigquery');
+
+const execSync = cmd => cp.execSync(cmd, {encoding: 'utf-8'});
+const generateUuid = () =>
+  `nodejs-samples-tests-routines-${uuid.v4()}`.replace(/-/gi, '_');
+const datasetId = generateUuid();
+const routineId = generateUuid();
+const newRoutineId = generateUuid();
+
+const bigquery = new BigQuery();
+
+describe('Routines', () => {
+  after(async () => {
+    await bigquery
+      .dataset(datasetId)
+      .delete({force: true})
+      .catch(console.warn);
```

Comment:
Do we have any concerns with leaking datasets here?  I'm ok with the `console.warn`, it's just unlikely it will ever be seen unless running locally.  What's the chance of leakage?

Reply:
@JustinBeckwith I think there's a really low chance of this failing. But we would leak datasets if the tests were stopped mid-run. An option to have a failed delete retried at some point could be to add labels to these testing datasets, then add logic to the before hook to find and delete any stale datasets with the label. This is the pattern used in the system tests and I think it makes sense to just add it to all the samples tests too.

Reply:
I added `deleteDatasets()` to the datasets test before hook, so leaked datasets should be cleaned up on subsequent runs.